compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -1116,16 +1116,13 @@ class Definitions {
       else
         FunctionType(args.length, isContextual).appliedTo(args ::: resultType :: Nil)
     def unapply(ft: Type)(using Context): Option[(List[Type], Type, Boolean)] = {
-      ft.dealias match
```

Comment:
What is the reason to drop the dealias in the extractor?

Reply:
There are a few reasons:
* There are few places where that type is not already dealiased
* In a few cases we used to repeatedly dealias the type in pattern match extractors for functions. Not dealiassing here forces the use site to dealias a single time. Sometimes this happens in cases of a single pattern match and sometimes in the recursion of a `TypeMap`.
* This aligns the implementation closer to `RefinedFunctionOf`, `PolyFunctionOf` and `FunctionTypeOfMethod` (in #18443).
* It brings us one step closer to being able to have a `derivedFunctionTypeOf` method.

Reply:
OK, I agree it's good to align the unapply's towards no dealising.